### DISCUSSION TITLE
Cannot install latest tailwind (>=3.0) in Rails 7

### DISCUSSION THREAD

**Discussion by kwokngakui** [rails/tailwindcss-rails#279] [Q&A]
I am new to Rails, and want to install latest tailwind v 3.3.3 in Rails 7.0.7 in CLI, however, it seems there is no any updated gems of version 3 available.

Do anyone know how to download the latest tailwind gem and install the gem locally? Or, where should I report to ask for updating tailwind to the latest version?

Thanks!!

P.S. Below is of what printed in CLI.

Fetching gem metadata from https://rubygems.org/...........
Could not find gem 'tailwindcss-rails (>= 3.0)' in rubygems repository https://rubygems.org/ or installed locally.

The source contains the following gems matching 'tailwindcss-rails':
.
.
.
  * tailwindcss-rails-2.0.30-aarch64-linux
  * tailwindcss-rails-2.0.30-arm-linux
  * tailwindcss-rails-2.0.30-arm64-darwin
  * tailwindcss-rails-2.0.30
  * tailwindcss-rails-2.0.30-x64-mingw-ucrt
  * tailwindcss-rails-2.0.30-x64-mingw32
  * tailwindcss-rails-2.0.30-x86_64-darwin
  * tailwindcss-rails-2.0.30-x86_64-linux

**Comment by flavorjones** [comment]
`tailwindcss-rails` v2.0.30 ships with Tailwind v3.3.3: https://github.com/rails/tailwindcss-rails/discussions/279

Please use the version of the **gem** and not the version of upstream tailwind. Thanks!

**Reply by kwokngakui** [reply]
Thanks for answering.

In fact, after installing the gem, and I copy some html text from Tailwind website into the rails app, it just doesn't work. I originally think that it is not Tailwind v3.3.3, because I read the v2 syntax is different from v3.

Now, I find out that the problem is: it doesn't work at all if I run command:

**rails server (or rails s)**

but it can work if I run command:

**./bin/dev**

I always do the **rails s**, since I learn rails. Do anyone know what is the difference between these two command?

**Comment by walterdavis** [comment]
Sure. When you have a JS backend process to do minification/tree-shaking/etc., that JS server needs to be started at the same time as the Rails server. bin/dev is just a wrapper around a process manager to start multiple servers at the same time. Note -- this is only needed in dev, because you don't want to be stopping your server, running `rake assets:precompile` and restarting each time you make a change. So the JS process watches for file changes, just like your Rails server watches for changes and reloads those updated files to get the new code.

Walter
